images.go

```diff
@@ -742,7 +744,28 @@ func (r *imageStore) Create(id string, names []string, layer, metadata string, c
 		list := r.bydigest[digest]
 		r.bydigest[digest] = append(list, image)
 	}
+	defer func() {
+		if err != nil {
+			// now that the in-memory structures know about the new
+			// record, we can use regular Delete() to clean up if
+			// anything breaks from here on out
+			if e := r.Delete(image.ID); e != nil {
```

Comment:
This won’t work with `image` a named return value.

Reply:
Fortunately we have a local variable with the same value in it (we even use it in the error message on the next line, doh), so the change needed isn't a large one.